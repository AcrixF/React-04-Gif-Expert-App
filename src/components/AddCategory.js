import React, { useState } from 'react';

const AddCategory = () => {

    const [inputValue, setInputValue] = useState('Brenda Marlen')

    const handleInputChange = (event) => {
        setInputValue(event.target.value);
    }

    const handleSubmit = (e) => {
        e.preventDefault();
        console.log(e);
    }

    return(
        <form onSubmit={handleSubmit}>
            <input 
                type="text" 
                value = {inputValue}
                onChange = { handleInputChange }/>
        </form>
    );
};

export default  AddCategory;