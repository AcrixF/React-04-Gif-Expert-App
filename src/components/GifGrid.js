import React, { useState, useEffect} from 'react';
import { GifGridItem } from './GifGridItem';

export const GifGrid = ({ category }) => {

    const [images, setimages] = useState([]);

     useEffect(() => {
        getGifs();
    }, [])

    const getGifs = async () => {
        const url = 'https://api.giphy.com/v1/gifs/search?api_key=K18TQHOfMMrCINMhsYf6BKbgeq99KP0z&q=Rick and Morty&limit=10';
        const response = await fetch(url);
        const { data } = await response.json();

        const gifs = data.map(img => {
            return {
                id: img.id,
                title: img.title,
                url: img.images?.downsized_medium.url
            }
        })
        console.log(data);

        setimages(gifs);
    }

    return(
    <>
        <h3>{ category }</h3>
         <div className="card-grid">
            {
                images.map((img) => 
                    <GifGridItem 
                        key={ img.id } 
                        {...img}
                    /> 
                )
            }
        </div>
    </>)
};

