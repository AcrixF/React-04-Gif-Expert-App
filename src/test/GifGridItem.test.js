import { shallow } from 'enzyme';
import {GifGridItem} from '../components/GifGridItem';
import React from 'react';

describe('should show the <GifGridItem/> correctly', () => {

    const title = 'Hunter x Hunter';
    const url = 'http:localhost/hunterxhunter.gif';
    const wrapper = shallow(<GifGridItem title={ title } url={ url }/>);

    test('should show the <GifGridItem/> shanpshot Correctly', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('should show the gifÂ´s title', () => {
        const pTitle = wrapper.find('p').text();
        expect(pTitle).toBe(title);
    });

    test('should have a img and an alt properties', () => {
        const img = wrapper.find('img');
        const source = img.prop('src');
        const alt = img.prop('alt');

        expect(source).toBe(url);
        expect(alt).toBe(title);
    });

    test('should validate <div></div> constains the className = card animate__animated animate__fadeInLeft', () => {
        const div = wrapper.find('div');
        
        expect(div.hasClass('card')).toBe(true);
        expect(div.hasClass('animate__animated')).toBe(true);
        expect(div.hasClass('animate__fadeInLeft')).toBe(true);
    })
    
    
    

});
