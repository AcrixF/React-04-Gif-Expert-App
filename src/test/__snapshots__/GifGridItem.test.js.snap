// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should show the <GifGridItem/> correctly should show the <GifGridItem/> shanpshot Correctly 1`] = `
<div
  className="card animate__animated animate__fadeInLeft"
>
  <img
    alt="Hunter x Hunter"
    src="http:localhost/hunterxhunter.gif"
  />
  <p>
    Hunter x Hunter
  </p>
</div>
`;
